manta-pay/concrete-protocol.tex

```diff
@@ -4,46 +4,85 @@
 
 \subsection{Poseidon Permutation}
 
-The \Poseidon{} Permutation \cite{KRRS21USENIX} is a finite field cryptographic primitive that can be used in lots of different contexts, like hash functions, commitment schemes, and symmetric encryption. \Poseidon{} plays a fundamental role in simplifying the protocol and reducing the overall cost of the Zero-Knowledge circuits. \Poseidon{} (without sponges) is a family of hash functions with the following signature:
-\[\Poseidon_k : \mathbb{F}^k \to \mathbb{F}\]
-over some sufficiencly large finite field $\mathbb{F}$. We make use of \Poseidon{} for a few values of $k$ in the concrete protocol below.
+The \Poseidon{} Permutation \cite{KRRS21USENIX} is a finite field cryptographic primitive that can be used in lots of different contexts, like hash functions, commitment schemes, and symmetric encryption. \Poseidon{} plays a fundamental role in simplifying the \Transfer{} protocol and reducing the overall cost of the Zero-Knowledge circuits. \Poseidon{} (without sponges) is a family of hash functions with the following signature:
+\[\Poseidon_k : \mathbb{F} \times \mathbb{F}^k \to \mathbb{F}\]
+over some sufficiencly large finite field $\mathbb{F}$. The first distinguished field element is used as a domain separation element. For this purpose, we use the following hashing function to generate domain strings:
+\[\HashToScalar(m) := \mathbb{F}.\truncate(\textsf{Blake2s}(m))\]
+
+We make use of \Poseidon{} for a few values of $k$ in the concrete protocol below.
 
 \subsection{Elliptic Curve Cryptography}
 
-Because we use a Zero-Knowledge Proving System, we want the cryptographic constructions that feature in our protocol to be \emph{ZKP-friendly}. For a ZKP system defined over a field $\mathbb{F}$ we can look for elliptic curves that have a base field of the same order as $\mathbb{F}$. These such curves are said to be ``embeddable'' or ``embedded in'' $\mathbb{F}$. For the constructions below, we use $\mathbb{F}$ as the proof system field and $\mathbb{G}$ as an embedded curve with scalar field $\mathbb{S}$. We also assume that $\left|\mathbb{S}\right| < \left|\mathbb{F}\right|$ so we can use the injection $\lift : \mathbb{S} \to \mathbb{F}$ to lift scalars to the proof system field.
+Because we use a Zero-Knowledge Proving System, we want the cryptographic constructions that feature in our protocol to be \emph{ZKP-friendly}. For a ZKP system defined over a field $\mathbb{F}$ we can look for elliptic curves that have a base field of the same order as $\mathbb{F}$. These such curves are said to be ``embeddable'' or ``embedded in'' $\mathbb{F}$. For the constructions below, we use $\mathbb{F}$ as the proof system field and $\mathbb{G}$ as an embedded curve with scalar field $\mathbb{S}$. We also assume that $\left|\mathbb{S}\right| < \left|\mathbb{F}\right|$ so we can use the injection $\lift : \mathbb{S} \to \mathbb{F}$ to lift scalars to the proof system field. 
```

Comment:
Two things:
- Add word finite in front of field. 
- All finite fields of the same order are isomorphic, so it doesn't make sense to say "that have a base field of the same order as F". Did you mean characteristic instead?